### PR TITLE
Add beta layouts

### PR DESCRIPTION
**Pull request by jklina**
This adds the beta `SplitPageLayout` and `PageLayout` components which will replace the existing alpha `Layout` component. 

This cannot be deployed until the corresponding CSS is first deployed: https://github.com/primer/css/pull/1737 and https://github.com/primer/css/pull/1849

### CODE REVIEW

File: demo/app/views/layouts/storybook_centered_preview.html.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <head>
     <title>ActionView::Component Storybook Preview</title>
 
-    <link href="https://unpkg.com/@primer/css/dist/primer.css" rel="stylesheet" />
+    <link href="https://unpkg.com/@primer/css@0.0.0-20211029165521/dist/primer.css" rel="stylesheet" />
```
Comment (jklina) [inline demo/app/views/layouts/storybook_centered_preview.html.erb:6]: Just using this to test against latest CSS. Will remove once CSS is merged in.

File: app/components/primer/beta/page_layout.rb [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+          @system_arguments[:classes] = class_names(
+            "PageLayout-region",
+            "PageLayout-content",
+            system_arguments[:classes]
+          )
+        end
+
+        def call
+          render(Primer::BaseComponent.new(**@system_arguments)) do
+            if @width == :fluid
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:414]:
```suggestion
            if @width == :full
```

File: app/components/primer/beta/page_layout.rb [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+        }.freeze
+        POSITION_OPTIONS = POSITION_MAPPINGS.keys.freeze
+
+        WIDTH_DEFAULT = :default
+        WIDTH_MAPPINGS = {
+          WIDTH_DEFAULT => "",
+          :narrow => "PageLayout--paneWidth-narrow",
+          :wide => "PageLayout--paneWidth-wide"
+        }.freeze
+        WIDTH_OPTIONS = WIDTH_MAPPINGS.keys.freeze
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:444]:
```suggestion
        WIDTH_DEFAULT = :medium
        WIDTH_MAPPINGS = {
          WIDTH_DEFAULT => "",
          :small => "PageLayout--paneWidth-narrow",
          :large => "PageLayout--paneWidth-wide"
        }.freeze
        WIDTH_OPTIONS = WIDTH_MAPPINGS.keys.freeze
```

File: app/components/primer/beta/page_layout.rb [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+          :start => "PageLayout--responsive-panePos-start",
+          :end => "PageLayout--responsive-panePos-end"
+        }.freeze
+        POSITION_NARROW_OPTIONS = POSITION_NARROW_MAPPINGS.keys.freeze
+
+        DIVIDER_DEFAULT = :start
+        DIVIDER_MAPPINGS = {
+          DIVIDER_DEFAULT => "PageLayout--panePos-start",
+          :end => "PageLayout--panePos-start"
+        }.freeze
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:458]: In the [CSS Storybook], there aren't "start" and "end" options for `divider`. Pane either has a divider or it doesn't

File: app/components/primer/beta/page_layout.rb [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+      #       <% c.pane_region(border: true) { "Pane" } %>
+      #       <% c.footer_region(border: true) { "Header" } %>
+      #     <% end %>
+      #     <%= render(Primer::Beta::PageLayout.new) do |c| %>
+      #       <% c.content_region(border: true) { "Content" } %>
+      #       <% c.pane_region(border: true) { "Pane" } %>
+      #       <% c.footer_region(divider: true, border: true) { "Header" } %>
+      #     <% end %>
+      #
+      # @param wrapper_sizing [Symbol] Define the maximum width of the component. `:fluid` sets it to full-width. Other values center Layout horizontally. <%= one_of(Primer::Beta::PageLayout::WRAPPER_SIZING_OPTIONS) %>
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:351]:
```suggestion
      # @param container_size [Symbol] Define the maximum width of the page container. `:full` sets it to full-width. Other values center Layout horizontally. <%= one_of(Primer::Beta::PageLayout::CONTAINER_WIDTH_OPTIONS) %>
```

File: app/components/primer/beta/page_layout.rb [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+        :filled => "PageLayout-region--dividerNarrow-filled-before"
+      }.freeze
+      FOOTER_DIVIDER_NARROW_OPTIONS = FOOTER_DIVIDER_NARROW_MAPPINGS.keys.freeze
+
+      # The layout's content.
+      #
+      # @param width [Symbol] <%= one_of(Primer::Beta::PageLayout::Content::WIDTH_OPTIONS) %>
+      # @param tag [Symbol] <%= one_of(Primer::Beta::PageLayout::Content::TAG_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :content_region, "Primer::Beta::PageLayout::Content"
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:89]:
```suggestion
      renders_one :content, "Primer::Beta::PageLayout::Content"
```

File: app/components/primer/beta/page_layout.rb [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+
+      # The layout's header.
+      #
+      # @param divider [Boolean] Whether to show a header divider
+      # @param divider_narrow [Symbol] Whether to show a divider below the `header` region if in responsive mode. <%= one_of(Primer::Beta::PageLayout::HEADER_DIVIDER_NARROW_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :header_region, lambda { |divider: false, divider_narrow: :line, **header_system_arguments|
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:96]:
```suggestion
      renders_one :header, lambda { |divider: false, divider_narrow: :line, **header_system_arguments|
```

File: app/components/primer/beta/page_layout.rb [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+
+        Primer::BaseComponent.new(tag: :div, **header_system_arguments)
+      }
+
+      # The layout's footer.
+      #
+      # @param divider [Boolean] Whether to show a footer divider
+      # @param divider_narrow [Symbol] Whether to show a divider below the `footer` region if in responsive mode. <%= one_of(Primer::Beta::PageLayout::FOOTER_DIVIDER_NARROW_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :footer_region, lambda { |divider: false, divider_narrow: FOOTER_DIVIDER_NARROW_DEFAULT, **footer_system_arguments|
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:113]:
```suggestion
      renders_one :footer, lambda { |divider: false, divider_narrow: FOOTER_DIVIDER_NARROW_DEFAULT, **footer_system_arguments|
```

File: app/components/primer/beta/page_layout.rb [hunk 9]
```diff
@@ -0,0 +1,504 @@
...
+      }
+
+      # The layout's pane.
+      #
+      # @param width [Symbol] <%= one_of(Primer::Beta::PageLayout::Pane::WIDTH_OPTIONS) %>
+      # @param position [Symbol] Pane placement when `Layout` is in column modes. <%= one_of(Primer::Beta::PageLayout::Pane::POSITION_OPTIONS) %>
+      # @param position_narrow [Symbol] Pane placement when `Layout` is in column modes. <%= one_of(Primer::Beta::PageLayout::Pane::POSITION_NARROW_OPTIONS) %>
+      # @param divider [Boolean] Whether to show a pane line divider.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :pane_region, lambda { |
```
Comment (colebemis) [inline app/components/primer/beta/page_layout.rb:134]:
```suggestion
      renders_one :pane, lambda { |
```

**Pull request closed without merging**